core/dbt/include/global_project/macros/materializations/tests/unit.sql

```diff
@@ -10,7 +10,7 @@
   {% set columns_in_relation = adapter.get_column_schema_from_query(get_empty_subquery_sql(sql)) %}
   {%- set column_name_to_data_types = {} -%}
   {%- for column in columns_in_relation -%}
-  {%- do column_name_to_data_types.update({column.name: column.data_type}) -%}
+  {%- do column_name_to_data_types.update({column.name|lower: column.data_type}) -%}
```

Comment:
This is naively assuming that the user has provided their columns in all lowercase. We'd obviously want to `lower` on both sides of the comparison.